pkg/deploy/taskdir/definitions/def_0_3.go

```diff
@@ -620,6 +644,30 @@ func (d *RESTDefinition_0_3) getConfigAttachments() []api.ConfigAttachment {
 	return configAttachments
 }
 
+// Rewrites Resource to be a slug if it's a name.
+func (d *RESTDefinition_0_3) normalize(ctx context.Context, client api.IAPIClient) error {
+	collection, err := getResourceIDsBySlugAndName(ctx, client)
+	if err != nil {
+		return err
+	}
+	slugsByID, err := getResourceSlugsByID(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	// Check slugs first.
+	if _, ok := collection.bySlug[d.Resource]; ok {
+		return nil
+	} else if id, ok := collection.byName[d.Resource]; ok {
+		if d.Resource, ok = slugsByID[id]; !ok {
+			return errors.Errorf("missing resource slug: %s", id)
```

Comment:
Clarify this error message. Same with REST.

```suggestion
			return errors.Errorf("could not find slug for resource with id: %q", id)
```